src/MongoDB/ReadPreference.c

```diff
@@ -153,9 +153,9 @@ static bool php_phongo_readpreference_init_from_hash(php_phongo_readpreference_t
 	return false;
 }
 
-static const char* php_phongo_readpreference_get_mode_string(mongoc_read_mode_t mode)
+static const char* php_phongo_readpreference_get_mode_string(const mongoc_read_prefs_t* read_prefs)
```

Comment:
This lets us eliminate some `mongoc_read_prefs_get_mode()` calls in parent contexts.

Additionally, I changed this function to return an empty string instead of throwing in the very unlikely event that the RP mode is invalid. [`_mongoc_read_mode_as_str()`](https://github.com/mongodb/mongo-c-driver/blob/1.28.0/src/libmongoc/src/mongoc/mongoc-read-prefs.c#L198) does the same, and libmongoc uses that for appending read preferences to command payloads. In our case, this is used for serialization and debugging, and the change allowed us to clean up the parent contexts.